src/main/java/kr/sesac/aoao/server/user/service/UserServiceImpl.java

```diff
@@ -2,27 +2,71 @@
 
 import static kr.sesac.aoao.server.user.exception.UserErrorCode.*;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
 import kr.sesac.aoao.server.global.exception.ApplicationException;
 import kr.sesac.aoao.server.user.controller.dto.request.SignUpRequest;
+import kr.sesac.aoao.server.user.domain.User;
+import kr.sesac.aoao.server.user.jwt.JwtTokenProvider;
+import kr.sesac.aoao.server.user.repository.UserEntity;
 import kr.sesac.aoao.server.user.repository.UserJpaRepository;
 import lombok.RequiredArgsConstructor;
 
+/**
+ * @since 2024.01.18
+ * @author 이상민
+ */
 @RequiredArgsConstructor
 @Transactional(readOnly = true)
 @Service
 public class UserServiceImpl implements UserService {
 
 	private final UserJpaRepository userRepository;
+	private final JwtTokenProvider jwtTokenProvider;
+	private final PasswordEncoder passwordEncoder;
 
```

Comment:
```suggestion
@RequiredArgsConstructor
@Transactional
@Service
public class UserServiceImpl implements UserService {

	private final UserJpaRepository userRepository;
	private final JwtTokenProvider jwtTokenProvider;
	private final PasswordEncoder passwordEncoder;

```
Class 레벨에는 `@Transactional` 만 붙이고, Method 레벨에는 `@Transactional(readOnly = true)` 를 붙이면 성능 상에 이점이 있다고 합니다!

서비스 계층에서 트랜잭션 과정이 필요하니 클래스 레벨에 붙였지만 그것과는 별개로 DB Replication 등의 작업을 적용한다면 readOnly 속성이 있을 때 Read 전용 데이터베이스 서버에 접근하여 읽기 속도가 향상된다고 합니다. ㅎ